### ISSUE TITLE
Exclude scripts from CloudFlare's Rocket Loader

### ISSUE THREAD

**Issue by aheckler** [Automattic/jetpack#1712]
On our [Known Issues page](http://jetpack.me/support/getting-started-with-jetpack/known-issues/), we discourage use of CloudFlare's Rocket Loader feature because it apparently interferes with some of our JS. We should just exclude our scripts from it instead:

https://support.cloudflare.com/hc/en-us/articles/200169436-How-can-I-have-Rocket-Loader-ignore-my-script-s-in-Automatic-Mode-

We'll need to obey the first two points in the "Notes" section.

**Comment by samhotchkiss** [comment]
This seems pretty straightforward, and I think we can get it in for this release.  I'm going to try to knock this out over the weekend.

**Comment by samhotchkiss** [comment]
Not quite so straightforward.

Either we can add in a bunch of filters, or we wait for this w.org issue to get resolved:
https://core.trac.wordpress.org/ticket/22249

@aheckler - do you know WHICH scripts have an issue with rocket loader?  If it's only one or two scripts, then we should go the filter method, it just seems unwieldy to add in filters for all of our JS

**Comment by aheckler** [comment]
> do you know WHICH scripts have an issue with rocket loader?

The only things I've found mention of in MGS that CloudFlare's Rocket Loader *for sure* breaks are Likes and Tiled Galleries. We also mention it breaking Stats on Jetpack.me's Known Issues list, but I can't find anything more about that.

> it just seems unwieldy to add in filters for all of our JS

Agreed, though I don't think we need to exempt *all* of our JS. CloudFlare [suggests not caching wp-admin](https://support.cloudflare.com/hc/en-us/articles/201717894-Using-CloudFlare-and-WordPress-Five-Easy-First-Steps) at all, so I'd think we'd only need to add filters to JS being output on the frontend.

**Comment by bhubbard** [comment]
I agree, a verified list needs to be compiled of what scripts need to filtered to exclude Rocket Loader. 

I don't think you can skip the admin scripts, because even though CloudFlare suggests not caching there is no guarantee a user will actually create that page rule. So for that reason I suggest we just automatically set every js file that loads in the wp-admin in the list of scripts that need the filter.

**Comment by lilmike** [comment]
Hey,
Is this still being worked on? I noticed that the likes are broken on my site, and I would like to see if this can be resolved.
-Michael.

**Comment by aheckler** [comment]
@lilmike Are you for sure that CF's Rocket Loader is the cause? There are a few different things that can break Likes, after all. Contact us here if you need assistance:

https://jetpack.com/contact-support/

**Comment by aheckler** [comment]
Still a thing.

**Comment by walidmujahid** [comment]
I just turned off rocket loader for a site to get the Likes working. I am wondering if this issue is still being worked on.
